src/cargo/util/config/value.rs

```diff
@@ -78,12 +78,7 @@ impl Definition {
     ///
     /// CLI is preferred over environment, which is preferred over files.
     pub fn is_higher_priority(&self, other: &Definition) -> bool {
-        match (self, other) {
-            (Definition::Cli, Definition::Environment(_)) => true,
-            (Definition::Cli, Definition::Path(_)) => true,
-            (Definition::Environment(_), Definition::Path(_)) => true,
-            _ => false,
-        }
+        matches!((self, other), (Definition::Cli, Definition::Environment(_)) | (Definition::Cli, Definition::Path(_)) | (Definition::Environment(_), Definition::Path(_)))
```

Comment:
I don't think this is really any more readable than before, and I might say it got worse, so could this be reverted?

Reply:
Ah, rustfmt failed to format this, I have filed https://github.com/rust-lang/rustfmt/issues/4512

Do you think this is acceptable?
````rust
    pub fn is_higher_priority(&self, other: &Definition) -> bool {
        matches!((self, other),
                 (Definition::Cli, Definition::Environment(_))
                 | (Definition::Cli, Definition::Path(_))
                 | (Definition::Environment(_), Definition::Path(_)))
    }
````

Reply:
I personally feel like the existing `match` was fine and more readable.  I like `matches!` for small things, but in this case it seems a bit awkward to me.

Reply:
Despite the semantics, `matches!` produces a fundamentally different AST structure that rustfmt works with, which in multi pattern cases ends up being a macro call expression with the second arg getting parsed as a binary expression. We do have a parser error with that second call arg in this case that occurs when there's both a semantic wildcard destructing in combination with the pattern join.

At some point down the road [we'll likely introduce](https://github.com/rust-lang/rustfmt/issues/4462#issuecomment-708432052) some special case formatting for `matches!` that would address this, but I wouldn't describe it as a top priority at this time. 

I personally like switching to `matches!` for the more trivial cases, but find the readability starts to drop quickly when the original `match` is less trivial.

If you want to use `matches!` here you'll need to format by hand, although will note that whenever we do get around to special case formatting it will almost certainly be different from the vertical alignment formatting proposed above.